### PR TITLE
Add Year to event URL

### PR DESCRIPTION
**Pull request by akirk**
By making the post type hierarchical, we can add the year to the slug.

WordCamp Europe in 2024 becomes https://translate.wordpress.org/events/2024/wordcamp-europe/ and in 2025 https://translate.wordpress.org/events/2025/wordcamp-europe/

add-year

### CODE REVIEW

File: includes/event/event-repository.php [hunk 1]
```diff
@@ -20,14 +20,36 @@ public function __construct( Attendee_Repository $attendee_repository ) {
 		$this->attendee_repository = $attendee_repository;
 	}
 
+	private function get_year_post_id( $year ) {
```
Comment (psrpinto) [inline includes/event/event-repository.php]:
```suggestion
        /*
         * @return int|WP_Error
         */
	private function get_year_post_id( string $year ) {
```

File: includes/event/event-repository.php [hunk 2]
```diff
@@ -20,14 +20,36 @@ public function __construct( Attendee_Repository $attendee_repository ) {
...
+		}
+
 		$event_id_or_error = wp_insert_post(
 			array(
 				'post_type'    => self::POST_TYPE,
 				'post_name'    => $event->slug(),
 				'post_title'   => $event->title(),
 				'post_content' => $event->description(),
 				'post_status'  => $event->status(),
+				'post_parent'  => $post_parent,
```
Comment (psrpinto) [inline includes/event/event-repository.php:57]: I think it would make sense to modify `test_create_event()` to make sure the post parent is set correctly.

**Review by psrpinto — APPROVED** [review]
Would it be possible to achieve this without needing a parent post, just by prepending `$year/` to the slug? Not sure that's possible. 

I was thinking that we would need to alter the `get_*()` methods of the repository to not return parent events, but WordPress already seems to filter them out somehow. WordPress must be checking if a post has children, and not return them in that case?

File: includes/event/event-repository.php (hunk 2 above)
Comment (akirk) [reply includes/event/event-repository.php:57]: Indeed, I'll add it!

**Comment by akirk** [comment]
> Would it be possible to achieve this without needing a parent post, just by prepending $year/ to the slug? Not sure that's possible.

Unfortunately not, slashes are not allowed in slugs. I also considered adding the year with a dash (like `2024-wordcamp-europe`) but it's more hassle to try and bend the generated slug vs just adding a parent post.

Using the hierarchy allows us to keep everything else unchanged and yet benefit from the "namespace separation."

**Pull request merged**